src/toolkits/object_detection/od_data_iterator.cpp

```diff
@@ -207,17 +211,16 @@ std::vector<image_annotation> parse_annotations(
   return result;
 }
 
-}  // namespace
-
-simple_data_iterator::annotation_properties
-simple_data_iterator::compute_properties(
-    const gl_sarray& annotations,
-    std::vector<std::string> expected_class_labels) {
-
-  annotation_properties result;
+std::pair<gl_sarray, size_t> get_annotation_info(const gl_sarray& annotations) {
+  if (annotations.size() == 0) {
+    return std::make_pair(gl_sarray(), 0);
+  }
 
   // Construct an SFrame with one row per bounding box.
   gl_sframe instances;
+  gl_sarray classes;
+  size_t num_instances;
```

Comment:
Why separate the declaration of these variables from their initialization?

Reply:
hmmmm ill fix this